### PR TITLE
Support R code formatter

### PR DESCRIPTION
**Pull request by dev-wei**
#44 

- [x] Add "formatr" code formatter 
- [x] Add a new Python command setting
- [x] Change visibility control for R/Python file
- [x] Version update

Aug-02-2019 15-21-00

### CODE REVIEW

**Comment by ryantam626** [comment]
That was quick, thanks for this! 😁

File: binder/environment.yml [hunk 1]
```diff
@@ -7,5 +7,6 @@ dependencies:
 - isort
 - jupyterlab=0.35
 - yapf
+- rpy2
 - pip:
   - jupyterlab_code_formatter
```
Comment (ryantam626) [inline binder/environment.yml]: I see that you removed new line in another commit (b043c1c25f13725deb6b318c925066aa4997745e), you can probably drop that commit - it was a mistake to not have new line that I didn't spot before and I am okay with having it fixed in the first commit (c9339e8d0d966e2583f219b0350a67cb0cb57892)

File: binder/environment.yml [hunk 2]
```diff
@@ -7,5 +7,6 @@ dependencies:
 - isort
 - jupyterlab=0.35
 - yapf
+- rpy2
```
Comment (ryantam626) [inline binder/environment.yml]: Maybe we should add `r-formatr` and whatever packages needed for a functional R environment in binder? (Not that fussed, you can opt not to do this if you don't want to)

Comment (dev-wei) [reply binder/environment.yml]: Is the binder will enforce them to be installed in the conda environment? r-formatr is available only from conda R repo.

File: binder/environment.yml (hunk 1 above)
Comment (dev-wei) [reply binder/environment.yml]: Good caught.

File: binder/environment.yml (hunk 2 above)
Comment (ryantam626) [reply binder/environment.yml]: > Is the binder will enforce them to be installed in the conda environment?
Hmm [binder](https://mybinder.org) is a tool we use to let user test out the plugin without installing it themselves, it doesn't the end user if they are installing it on their machine.
> r-formatr is available only from conda R repo
I think we can add the r channel in the channels section of `environment.yml` and it should be fine, we are already using `conda-forge` which is not the default channel.

Comment (dev-wei) [reply binder/environment.yml]: Make sense.

**Comment by dev-wei** [comment]
@ryantam626  Been busy for last couple days. Please review the changes and let me know for any extra feedback.

FYI, I am working on another PR for supporting tidyverse styler for R. https://github.com/r-lib/styler, i really love their non-invasive formatting for tidyverse code.

**Comment by ryantam626** [comment]
Oh cool! I will try to get this merged and published tonight :grin: Thanks again!

**Comment by dev-wei** [comment]
You bet!

**Comment by dev-wei** [comment]
@ryantam626  any update on this PR?

**Comment by ryantam626** [comment]
Hey! Sorry for being afk, my other projects required my immediate attention :scream: gonna do it now

**Pull request merged**